service/src/main/java/uk/nhs/adaptors/gp2gp/tasks/TaskDefinition.java

```diff
@@ -0,0 +1,18 @@
+package uk.nhs.adaptors.gp2gp.tasks;
+
+import static java.util.UUID.randomUUID;
+
+import lombok.Getter;
+
+@Getter
+public abstract class TaskDefinition {
+    private final String taskId;
+    private final String requestId;
+    private final String conversationId;
+
+    public TaskDefinition(String requestId, String conversationId) {
+        this.taskId = randomUUID().toString();
```

Comment:
This will cause a bug. What happens when a TaskDefinition is deserialised from JSON? It will have a random taskId instead of the taskId from the JSON document.

Assigning the id is effectively application logic and should not be part of a POJO / model.